EasyPost.Tests/InsuranceTest.cs

```diff
@@ -0,0 +1,145 @@
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            ClientManager.SetCurrent("NvBX2hFF44SVvTPtYjF0zQ");
+            _fromAddressData = new Dictionary<string, object>
+            {
+                {
+                    "company", "Simpler Postage Inc"
+                },
+                {
+                    "street1", "164 Townsend St"
+                },
+                {
+                    "street2", "Unit 1"
+                },
+                {
+                    "city", "San Francisco"
+                },
+                {
+                    "state", "CA"
+                },
+                {
+                    "zip", "94107"
+                },
+                {
+                    "country", "US"
+                }
+            };
+            _toAddressData = new Dictionary<string, object>()
+            {
+                {
+                    "company", "The White House"
+                },
+                {
+                    "street1", "1600 Pennsylvania Avenue NW"
+                },
+                {
+                    "street2", "DC 20500"
+                },
+                {
+                    "city", "Washington"
+                },
+                {
+                    "state", "DC"
+                },
+                {
+                    "zip", "20500"
+                },
+                {
+                    "country", "US"
+                }
+            };
+            _parcelData = new Dictionary<string, object>
+            {
+                {
+                    "weight", 8.0
+                }
+            };
+
+            Shipment shipment = Shipment.Create(new Dictionary<string, object>
+            {
+                {
+                    "from_address", _fromAddressData
+                },
+                {
+                    "to_address", _toAddressData
+                },
+                {
+                    "parcel", _parcelData
+                },
+                {
+                    "service", "First"
+                },
+                {
+                    "carrier_accounts", new List<string>
+                    {
+                        {
+                            "ca_7642d249fdcf47bcb5da9ea34c96dfcf"
+                        }
+                    }
+                }
+            });
+
+            InsuranceData = new Dictionary<string, object>
+            {
+                {
+                    "to_address", _toAddressData
+                },
+                {
+                    "from_address", _fromAddressData
+                },
+                {
+                    "tracking_code", shipment.tracking_code
+                },
+                {
+                    "amount", (float)100
+                },
+                {
+                    "carrier", "USPS"
+                }
+            };
+        }
+
+        [TestMethod]
+        public void TestCreateAndRetrieve()
```

Comment:
Longterm, tests should only do one thing (test one thing) - We can leave this for now but when we circle back, there should be a standalone `create` test, a standalone `retrieve` test (which may still need to create an object, but it shouldn't assert against the created one, only the retrieved one), and then an `all` test aside.

Reply:
Almost every one of our test files has a `CreateAndRetrieve` method, so we'll need to rework that for the entire library and all tests.